src/extension/sldService/src/main/java/org/geoserver/sldservice/utils/classifier/RasterSymbolizerBuilder.java

```diff
@@ -496,7 +496,9 @@ private String getPercentagesLabelPortion(double[] percentages, int i) {
         double total = DoubleStream.of(classMembers).sum();
         double[] percentages = new double[classNum];
         for (int i = 0; i < classNum; i++) {
-            percentages[i] = (classMembers[i] / total) * 100;
+            double cm = classMembers[i];
+            if (cm != 0d && total != 0d) percentages[i] = (classMembers[i] / total) * 100;
```

Comment:
You extrac cm, but then use classMembers[i] right next line? :-D

```suggestion
            if (cm != 0d && total != 0d) percentages[i] = (cm / total) * 100;
```